runtime/interpreter/number_test.go

```diff
@@ -166,3 +167,89 @@ func TestOverEstimateFixedPointStringLength(t *testing.T) {
 		)
 	}
 }
+
+func BenchmarkValueArithmetic(b *testing.B) {
+
+	b.ReportAllocs()
+
+	isNegatable := func(v NumberValue) bool {
+		switch v.(type) {
+		case IntValue, Int8Value, Int16Value,
+			Int32Value, Int64Value, Int128Value,
+			Int256Value, Fix64Value:
+			return true
+		}
+		return false
+	}
```

Comment:
i think you can replace this with a check if v is a subtype of `SignedNumberValue`

Reply:
What type is this? I don't think this actually exists in the interpreter.